src/java.base/share/classes/sun/security/ssl/SSLTransport.java

```diff
@@ -122,16 +128,48 @@ static Plaintext decode(TransportContext context,
         } catch (AEADBadTagException bte) {
             throw context.fatal(Alert.BAD_RECORD_MAC, bte);
         } catch (BadPaddingException bpe) {
-            /*
-             * The basic SSLv3 record protection involves (optional)
-             * encryption for privacy, and an integrity check ensuring
-             * data origin authentication.  We do them both here, and
-             * throw a fatal alert if the integrity check fails.
-             */
-             Alert alert = (context.handshakeContext != null) ?
-                     Alert.HANDSHAKE_FAILURE :
-                     Alert.BAD_RECORD_MAC;
-            throw context.fatal(alert, bpe);
+            // Check for unexpected plaintext alert message during TLSv1.3+ handshake.
+            // This can happen if client doesn't receive server_hello due to network timeout
+            // and tries to close the connection by sending an alert message.
+            if (context.inputRecord instanceof SSLSocketInputRecord) {
+                currentFlight = context.inputRecord.getLastDecodeRecord();
+            }
+
+            if (currentFlight != null &&
+                    !context.sslConfig.isClientMode &&
+                    !context.isNegotiated &&
+                    context.handshakeContext != null &&
+                    context.handshakeContext.negotiatedProtocol != null &&
+                    context.handshakeContext.negotiatedProtocol.useTLS13PlusSpec()) {
+
+                byte contentType = (byte) Record.getInt8(currentFlight);             // pos: 0
+                byte majorVersion = (byte) Record.getInt8(currentFlight);            // pos: 1
+                byte minorVersion = (byte) Record.getInt8(currentFlight);            // pos: 2
+                int contentLen = Record.getInt16(currentFlight);                     // pos: 3, 4
+
+                if (contentLen == 2 && ContentType.ALERT.equals(ContentType.valueOf(contentType))) {
```

Comment:
The timeout case could happen during rekeying as well.  For rekeying, the content might be encrypted with previous keys and cannot decrypted with the current keys.  As may void the condition "contentLen == 2".

Reply:
KEY UPDATE happens after handshake, so `!context.isNegotiated` check above won't pass.